### PR TITLE
Add service module for Attic

### PR DESCRIPTION
**Pull request by pizzapim**
The example generates the server token and a user token to quickly try out the server.

Try it out with:
```
$ nix build '.#examples.attic.config.system.build.ociImage.stream' && ./result | docker load && docker run -it -p 8080:8080 nixng-attic
```

And then:
```
$ attic login demo http://localhost:8080 $token
$ attic cache create demo:mycache
$ attic push demo:mycache $(nix build --no-link --print-out-paths '.#examples.attic.config.system.build.ociImage.build')
```

### CODE REVIEW

**Comment by MagicRB** [comment]
Looks good so far, just a question tho. Doesn't attic need some state directory? If so then ensuring its existence using the `ensureSomething`s system would be nice. (I hate the name, why did I think that was a good name....)

File: modules/services/attic.nix [hunk 1]
```diff
@@ -23,6 +23,18 @@ in
     settings = lib.mkOption {
       type = lib.types.submodule {
         freeformType = settingsFormat.type;
+
+        options = {
+          storage = {
+            type = lib.mkOption {
+              type = lib.types.str;
```
Comment (MagicRB) [inline modules/services/attic.nix:30]: `type` should be an `enum` i think. And `path` can be `types.path`

Comment (pizzapim) [reply modules/services/attic.nix:30]: Type can be `s3` (or something) as well so we probably shouldn't restrict `path` to `types.path`. But it should probably be `enum` yeah.

**Comment by pizzapim** [comment]
Just realized it would actually be better to do the config checks at build-time if possible. Same applies to the dnsmasq service.

**Comment by MagicRB** [comment]
I havent been doing that, couldnt be bothered but yes, if possible it would be nice. Good catch!

File: modules/services/attic.nix [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+        freeformType = settingsFormat.type;
+
+        options = {
+          storage = {
+            type = lib.mkOption {
+              type = lib.types.enum [ "local" "s3" ];
+            };
+
+            path = lib.mkOption {
+              type = lib.types.str;
```
Comment (MagicRB) [inline modules/services/attic.nix:34]: This could be `types.path`

**Comment by MagicRB** [comment]
Is there no way to pass the credentials to attic without the use of an environment variable? Environment variables are not the most secure thing ever

**Comment by pizzapim** [comment]
Looking at [the code](https://github.com/zhaofengli/attic/blob/main/server/src/config.rs#L111), it can only be set either in the config file directly or using an environmental variable.

File: modules/services/attic.nix (hunk 2 above)
Comment (pizzapim) [reply modules/services/attic.nix:34]: Like I said above, the type can be `s3` as well so we cannot restict the path to be a file system path only.

Comment (MagicRB) [reply modules/services/attic.nix:49]: Oh sorry that I missed.

**Comment by MagicRB** [comment]
Okay then no need to screw around with config file merging. Thats waay too much work.

**Pull request merged**